mangaki/mangaki/utils/recommendations.py

```diff
@@ -5,119 +5,142 @@
 from django.contrib.auth.models import User
 from django.db.models import Count
 from django.db import connection
+import random
+from datetime import datetime
+from mangaki.utils.wals import MangakiWALS
+from mangaki.utils.als import MangakiALS
+from mangaki.utils.knn import MangakiKNN
+from mangaki.utils.svd import MangakiSVD
+import numpy as np
+import pickle
+import os.path
+
+
+NB_NEIGHBORS = 20
+NB_RECO = 10
+RATED_BY_AT_LEAST = 2
+CHRONO_ENABLED = False
+
+
+def make_anonymous_data(queryset):
+    triplets = []
+    users = set()
+    works = set()
+    nb_ratings = Counter()
+    X = []
+    y = []
+    for user_id, work_id, rating in queryset.values_list('user_id', 'work_id', 'choice'):
+        users.add(user_id)
+        works.add(work_id)
+        triplets.append((user_id, work_id, rating))
+        nb_ratings[work_id] += 1
+    random.shuffle(triplets)  # Scramble time
+
+    anonymous_u = list(range(len(users)))
+    anonymous_w = list(range(len(works)))
+    random.shuffle(anonymous_u)
+    random.shuffle(anonymous_w)
+    encode_user = dict(zip(users, anonymous_u))
+    encode_work = dict(zip(works, anonymous_w))
+    decode_user = dict(zip(anonymous_u, users))
+    decode_work = dict(zip(anonymous_w, works))
+
+    interesting_works = set()
+    for work_id, _ in nb_ratings.most_common():
+        if nb_ratings[work_id] < RATED_BY_AT_LEAST:
+            break
```

Comment:
Why `break` here?
Could you explain more about this part? Is it because anyway, the next `nb_ratings[work_id]` will be rated by a lesser amount of users?